tests/__init__.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+
+from typing import Callable
+
+
+def check(actual: object, klass: type, dtype: type | None = None) -> None:
+
+    if not isinstance(actual, klass):
+        raise RuntimeError(f"Expected type '{klass}' but got '{type(actual)}'")
+    if dtype is None:
+        return None
+
+    if hasattr(actual, "__iter__"):
+        value = next(iter(actual))  # type: ignore[call-overload]
+    else:
+        value = actual.left  # type: ignore[attr-defined]
```

Comment:
The `.left` attribute is arbitrary.  So could add parameter to `check` which is `attr: Optional[str] = None` and then do:
```python
if attr is None:
    value = next(iter(actual)) # type: ignore[call-overload]
else:
    value = actual.__getattr__(attr)
```
Then the caller can decide which attribute to check.

Reply:
Added the `attr` argument. I gave it the default value `"left"` to keep the `check(...)` calls short.